### PR TITLE
Implement secure HTTPS communication. 

### PR DESCRIPTION
**Pull request by bhaskarvilles**
Consider using latest TLSv1.2 instead of TLS.

Signed-off-by : bhaskarvilles <bhaskarvilles@yandex.com>

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 1]
```diff
@@ -2877,7 +2877,7 @@ protected void enableGoQuorumCompatibilityMode() {
 
   public static List<String> getJDKEnabledCypherSuites() {
     try {
-      SSLContext context = SSLContext.getInstance("TLS");
+      SSLContext context = SSLContext.setInstance("TLS 1.2");
```
Comment (frankisawesome) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2880]: It looks like we're also changing the method call from `getInstance` to `setInstance` here, is that expected?

**Review by frankisawesome — CHANGES_REQUESTED** [review]
You will also need to make sure every single commit of the PR is signed off using `git commit -s`, check the DCO step on CI for a guide on how to amend your latest commit.

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 1 above)
Comment (pinges) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2880]: We should specify TLS 1.3 here. TLS 1.3 is the latest, best performing, and most secure version of TLS. TLS 1.3 means that we are allowing TLS 1.3 and lower, but 1.3 is the preferred one!
I don't think there is a setInstance method on SSLContext!

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java [hunk 2]
```diff
@@ -2888,7 +2888,7 @@ protected void enableGoQuorumCompatibilityMode() {
 
   public static List<String> getJDKEnabledProtocols() {
     try {
-      SSLContext context = SSLContext.getInstance("TLS");
+      SSLContext context = SSLContext.setInstance("TLS 1.2");
```
Comment (pinges) [inline besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2891]: see my comment above.

**Review by pinges — CHANGES_REQUESTED** [review]

**Comment by macfarla** [comment]
since there's only one commit this should do it for the DCO @bhaskarvilles 
  git commit --amend --no-edit --signoff

**Comment by bhaskarvilles** [comment]
> since there's only one commit this should do it for the DCO @bhaskarvilles git commit --amend --no-edit --signoff

Invited for Collaboration please pass DCO checks.

**Comment by macfarla** [comment]
@bhaskarvilles it's still failing DCO on commit https://github.com/hyperledger/besu/pull/3380/commits/a1fa203f3831d7c5d4ce08957b916a52d04b3ef2 
:/

**Comment by macfarla** [comment]
@bhaskarvilles are you able to fix the DCO and the other requested changes on this PR?

**Comment by bhaskarvilles** [comment]
> @bhaskarvilles are you able to fix the DCO and the other requested changes on this PR?

I don't know what i did wrong in configuring the DCO, not passing Checks :( wait I'll invite you as collaborators and the you can check the issue ?!

File: besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java (hunk 2 above)
Comment (bhaskarvilles) [reply besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java:2901]: I'm sorry for late reply I've forgotten to change the setInstance method to getInstance and TLS support setInstance also, I'm changing to v1.3 now :), rebasing branch :)

**Comment by frankisawesome** [comment]
It might be easier if you delete your local branch, start a brand new branch off latest master, do the correct changes to 1.3 and then force push to this remote branch, and make sure you have -s on git commit

**Comment by macfarla** [comment]
This is the commit without signoff @bhaskarvilles https://github.com/hyperledger/besu/pull/3380/commits/dfba826baf5e9a454841e9c7a21d7d0785746f17

**Pull request closed without merging**